### DISCUSSION TITLE
Does this work with 2FA?

### DISCUSSION THREAD

**Discussion by ZoltrixGFC** [dh024/Bitwarden_Export#4] [Q&A] [answered]
Hi,

Thanks for the tip on the forums regarding the script and Windows 👍

Now when running the script it gives me, "ERROR: Failed to authenticate". I thought I better double check, does this script work if you have 2FA enabled on your Bitwarden account?

**Comment by dh024** [comment, marked as answer]
Yes, it will work with authenticator codes, email, or recovery codes. It might also work with Yubikey and Duo - the Bitwarden help is not clear on this, and I don't use either to test. It will not work with FIDO2 WebAuthn, however.

If you received a "Failed to authenticate" error, it means something went wrong with the login process, so either the wrong username (email) or password was entered. Did you set your email within the script at line 15?

**Reply by ZoltrixGFC** [reply]
Yes, have definitely entered my Bitwarden email address in the config, full output as follows:

Starting export script...
Enter your Bitwarden password:
new.sh: line 44: bw: command not found
new.sh: line 44: jq: command not found
new.sh: line 49: bw: command not found
new.sh: line 49: jq: command not found
new.sh: line 57: bw: command not found
ERROR: Failed to authenticate.

**Reply by dh024** [reply]
You really should review and follow the information provided in the README file here, particularly the Requirements section:

https://github.com/dh024/Bitwarden_Export#readme

**Reply by ZoltrixGFC** [reply]
I have read the instructions. Just trying to provide you feedback on your script. I'm happy to stop here if you don't have the time to investigate... the configurable parts of the script I have had configured since the first time I attempted to use it, and it looks as follows:

<img width="926" alt="Screenshot 2022-09-12 143910" src="https://user-images.githubusercontent.com/77313738/189575331-4a1b8f6e-b457-4815-8510-08443470806b.png">

**Reply by dh024** [reply]
The script is not the issue you are experiencing. It is trying to call the Bitwarden CLI and the jq tool, neither of which are available to it.

**Reply by ZoltrixGFC** [reply]
Hang on a minute, did you just add the "Requirements" section to the documentation? 😜

Apologies if it was already there, I completely missed it, and it is clearly states the additional components needed.
